src/support/err.c

```diff
@@ -504,12 +502,11 @@ __wt_illegal_value(WT_SESSION_IMPL *session, const char *name)
 	    name == NULL ? "" : name, name == NULL ? "" : ": ",
 	    "encountered an illegal file format or internal value");
 
-#if !defined(HAVE_DIAGNOSTIC)
-	return (__wt_panic(session));
+#if defined(HAVE_DIAGNOSTIC)
+	__wt_abort(session);                    /* Drop core if testing. */
 #endif
 
-	__wt_abort(session);			/* Drop core if testing. */
-	/* NOTREACHED */
+	return (WT_ERROR);
```

Comment:
> Yes and no. If we can't open a file and read the first block, I agree we want to fail gracefully, it's likely a configuration/encryption problem. If we're running for a day and suddenly a block read fails, I think it's a pretty big problem and we want to shut down fast, before more damage occurs.

That's why I feel that panic should be handled at the top level rather than deep down in a macro like `WT_ILLEGAL_VALUE` -- if instead it returned a `WT_ILLEGAL_VALUE` error (or just `EINVAL`), we can decide above whether that warrants a panic or not.  There is no way for the low-level code to know the difference between the two situations short of more special session flags.